File: convert.js
const tf = require('@tensorflow/tfjs-node');
const fs = require('fs');

const loadModel = async () => {
  const modelPath = './models/model.json';
  const model = await tf.loadLayersModel(`file://${modelPath}`);
  return model;
};

// Carregar o modelo
const model = await loadModel();

// Realizar previsões com o modelo
const input = tf.tensor2d([[...seus_dados_de_entrada]]);
const predictions = model.predict(input);

// Converter as previsões para um array
const predictionsArray = await predictions.array();

console.log(predictionsArray);

File: convert.py
import tensorflow as tf
from keras.models import load_model

# Definir a função de perda personalizada
def sparse_cat_loss(y_true, y_pred):
    # Implemente sua função de perda aqui
    pass

# Registrar a função de perda personalizada
tf.keras.utils.get_custom_objects()['sparse_cat_loss'] = sparse_cat_loss

# Carregar o modelo salvo em formato H5
model = load_model('luizgonzaga_gen_mjr.h5')

# Salvar o modelo como JSON
model_json = model.to_json()
with open('./models/model.json', 'w') as json_file:
    json_file.write(model_json)
